File: String/2299.Strong Password Checker II/Solution.py
class Solution:
    def strongPasswordCheckerII(self, password: str) -> bool:
        if len(password) < 8:
            return False
        sm = bi = num = sp = False
        last = ""
        for c in password:
            if c == last:
                return False
            if 'a' <= c <= 'z':
                sm = True
            if 'A' <= c <= 'Z':
                bi = True
            if '0' <= c <= '9':
                num = True
            if c in "!@#$%^&*()-+":
                sp = True
            last = c
        return sm and bi and num and sp

File: String/2299.Strong Password Checker II/Solution.java
class Solution {
    public boolean strongPasswordCheckerII(String password) {
        int n = password.length();
        if (n < 8) {
            return false;
        }
        boolean sm = false, bi = false, num = false, sp = false;
        char last = ' ';
        for (char c : password.toCharArray()) {
            if (c == last) {
                return false;
            }
            if (c >= 'a' && c <= 'z') {
                sm = true;
            }
            if (c >= 'A' && c <= 'Z') {
                bi = true;
            }
            if (c >= '0' && c <= '9') {
                num = true;
            }
            if ("!@#$%^&*()-+".indexOf(c) != -1) {
                sp = true;
            }
            last = c;
        }
        return sm && bi && num && sp;
    }
}